### PR TITLE
docs: created a new document for Provisioning NFS PVCs

### PR DESCRIPTION
**Pull request by balaharish7**

### CODE REVIEW

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+  **Command**
+  
+  ```
+  $ kubectl create ns nfs-server
+  ```
+
+  **Output**
+  
+  ```
+  $ kubectl create ns nfs-server
```
Comment (datacore-tilangovan) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:116]: remove command from here.

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+  namespace/nfs-server created
+  ```
+
+4. Create a PVC with a desired storage for the NFS server using the Replicated PV Mayastor storage class.
+
+  **Command**
+
+  ```
+  $ cat <<EOF | kubectl create -f -
```
Comment (datacore-tilangovan) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:125]: The command is wrong here, please refer the confluence for the command.

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+    namespace: nfs-server
+  spec:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 5Gi
+    storageClassName: mayastor-3
+  EOF
+  persistentvolumeclaim/nfs-server-claim created
```
Comment (datacore-tilangovan) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:144]: Only this line is output rest all are command itself. refer confluence for the same.

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md (hunk 2 above)
Comment (balaharish7) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:125]: image
This is the command given in the confluence page.

Comment (datacore-tilangovan) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:125]: Nope, thats just a beginning of the command. 

This is one single command as 
cat <<EOF | kubectl create -f -
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: nfs-server-claim
  namespace: nfs-server
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
  storageClassName: mayastor-3
EOF

Comment (datacore-tilangovan) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:125]: This is one single command, the one you have above is just a beginning of the command. you can also see this from openebs official doc. 

image

**Review by datacore-tilangovan — APPROVED** [review]

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/gke.md [hunk 4]
```diff
@@ -313,4 +313,5 @@ root@nginx-deployment-7bf66b59f5-mxcdp:/volume#
 ## See Also
 
 - [Replicated PV Mayastor Installation on MicroK8s](microkubernetes.md)
-- [Replicated PV Mayastor Installation on Talos](talos.md)
\ No newline at end of file
+- [Replicated PV Mayastor Installation on Talos](talos.md)
+- [Provisioning Read-Write-Many (RWX) PVCs](rwx.md)
```
Comment (avishnu) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/gke.md:317]: How is this relevant here?

Comment (avishnu) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/gke.md:317]: I think this is not relevant under this folder/group. It needs a different group which can include
- rwx
- object store (in the future, when we solution with minio)

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+
+# Provisioning Read-Write-Many (RWX) PVCs
+
+Provisioning Read-Write-Many (RWX) Persistent Volume Claims (PVCs) involves setting up storage that can be simultaneously accessed in read-write mode by multiple nodes or pods within a Kubernetes cluster.
+
+Block storage, such as Persistent Volumes (PVs) in Kubernetes, is generally not designed to be mounted to multiple pods simultaneously due to several technical limitations and potential issues:
+
+- **File System Consistency**
+
+  Block storage devices are designed for exclusive access. Mounting the same block device to multiple pods can lead to file system corruption due to concurrent writes and uncoordinated access. Traditional file systems like ext4 or XFS are not designed to handle multiple writers, leading to inconsistencies and corruption.
```
Comment (avishnu) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:20]: should it be uncordinated or non-co-ordinated?

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md [hunk 6]
```diff
@@ -0,0 +1,496 @@
...
+
+- **Performance Issues**
+
+  Concurrent access by multiple pods can lead to performance degradation. The storage system might not be optimized for handling simultaneous I/O operations from multiple sources, causing bottlenecks and reduced performance.
+
+OpenEBS Replicated PV Mayastor supports ReadWriteMany (RWX) volumes by exposing regular Replicated PV Mayastor volumes via NFSv4 servers as a pod.
+
+## Concept
+
+Replicated PV Mayastor uses the NFS (Network File System) server as a deployment with the NFS CSI driver. This provides PVCs in RWX mode so that multiple web applications can access the data in a shared fashion. Replicated PV Mayastor volumes are used as persistent backend storage for these NFS servers to provide a scalable and manageable RWX shared storage solution.
```
Comment (avishnu) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:38]: I think we can remove "web"

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+
+  When multiple pods attempt to read from and write to the same block storage simultaneously, it can cause conflicts and data corruption. Block storage lacks the necessary coordination mechanisms to manage concurrent access, which is why shared access is problematic.
+
+- **Lack of Coordinated Locking**
+
+  File systems on block storage rely on locking mechanisms to ensure data integrity during concurrent access. These locks are not shared across different instances of the file system, leading to the risk of data corruption when the same block storage is mounted to multiple pods.
+
+- **Performance Issues**
+
+  Concurrent access by multiple pods can lead to performance degradation. The storage system might not be optimized for handling simultaneous I/O operations from multiple sources, causing bottlenecks and reduced performance.
```
Comment (avishnu) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:32]: The above 4 sections tend to downplay the importance of block storage which I think we shouldn't be doing. We just need to highlight the use-cases of rwx, not try to downgrade block.

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md [hunk 8]
```diff
@@ -0,0 +1,496 @@
...
+```
+
+**Output**
+
+```
+NAME          STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
+nfs-claim-1   Bound    pvc-f795f8bc-58b3-40d7-b9b2-2d16c17f3d47   5Gi        RWX            nfs-csi        <unset>                 10s
+```
+
+2. Deploy the application.
```
Comment (avishnu) [inline docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:443]: Is nginx the right example here, @datacore-tilangovan ?
I think, we also need to show-case rwx use-case here, by deploying more than one pod/application using the shared storage.

Comment (datacore-tilangovan) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:443]: we have nginx with three replicas here using pvc (nfs-claim-1).

we can use standalone Db's as example, but keeping them replicated as deployments is not ideal use case as they will be deployed as statefulsets. so we have used nginx here.

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md (hunk 7 above)
Comment (datacore-tilangovan) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:32]: Agreed ! sure, we can change this.

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/gke.md (hunk 4 above)
Comment (balaharish7) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/gke.md:317]: @avishnu Can we create a folder named "Operations" to place the RWX docs?

Comment (avishnu) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/gke.md:317]: I think, Read Write Many will be better for the folder name. For the current topic, we could call "Provisioning NFS PVCs"

File: docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md (hunk 8 above)
Comment (avishnu) [reply docs/main/Solutioning/openebs-on-kubernetes-platforms/rwx.md:443]: ok
